### PR TITLE
Make sure countdown component works when there's no package context

### PR DESCRIPTION
**Pull request by vegaro**
Makes the countdown component independent from package context, allowing countdown variables to be processed even when no package is available. It's an edge case but it can happen easily in unpublished paywalls.

Adds countdown previews to CountdownComponentView for both running and times-up states and also tests

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/text/TextComponentView.kt [hunk 1]
```diff
@@ -112,41 +110,34 @@ private fun rememberProcessedText(
...
-                val dateLocale = state.locale.toJavaLocale()
-                val currencyLocale = state.currencyLocale.toJavaLocale()
-
-                val introEligibility = packageToUse.introEligibility
-
-                when (introEligibility) {
-                    IntroOfferEligibility.INELIGIBLE -> textState.text
-                    IntroOfferEligibility.SINGLE_OFFER_ELIGIBLE -> textState.text
-                    IntroOfferEligibility.MULTIPLE_OFFERS_ELIGIBLE -> textState.text
-                }
```
Comment (vegaro) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/text/TextComponentView.kt:125]: This was dead code. I honestly don't know what it was supposed to do. I think it was attempting to return the text without variables?

Comment (JayShortway) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/text/TextComponentView.kt:125]: I see I added this in https://github.com/RevenueCat/purchases-android/pull/2058 a year ago. I really don't remember what the idea was here haha. Are we sure the text component view properly handles intro offer eligibility?

Comment (vegaro) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/text/TextComponentView.kt:125]: yeah I need to confirm. I assume it is correctly handled otherwise we would have heard of it, but will double check

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt [hunk 2]
```diff
@@ -277,156 +275,138 @@ internal object VariableProcessorV2 {
...
+        Variable.PRODUCT_SECONDARY_OFFER_PERIOD ->
+            rcPackage?.secondIntroOffer?.productOfferPeriod(localizedVariableKeys)
 
-        Variable.PRODUCT_RELATIVE_DISCOUNT -> packageContext.relativeDiscount(localizedVariableKeys)
+        Variable.PRODUCT_SECONDARY_OFFER_PERIOD_ABBREVIATED ->
+            rcPackage?.secondIntroOffer?.productOfferPeriodAbbreviated(localizedVariableKeys)
 
-        Variable.PRODUCT_STORE_PRODUCT_NAME -> rcPackage.product.name
+        Variable.PRODUCT_RELATIVE_DISCOUNT -> packageContext?.relativeDiscount(localizedVariableKeys)
+        Variable.PRODUCT_STORE_PRODUCT_NAME -> rcPackage?.product?.name
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt:409]: I wonder if we should split the variables with and without package support for cleanliness, since now most variables need to handle nullability... I'm assuming the backend already would do some of the validations to make sure we only use these variables when appropriate... But not a blocker, I would say.

**Review by tonidero — APPROVED** [review]
Makes sense, thanks for fixing this!! 🙏

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt (hunk 2 above)
Comment (vegaro) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt:409]: yeah, I tried that approach, but first checking the type of variable with a `when`, then running the replacement of the variables with another `when`. But thinking about it now... maybe we can have two sets, and we just need to check the variable is part of either set. That should be pretty efficient. Let me try that approach

Comment (vegaro) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt:409]: Actually I am not convinced. I think it can become fragile when someone adds a new variable...

I think I am just going to leave it as is.

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt:409]: Sounds good to me! I do think it's nice to have all variables at the same level 👍
